pkg/engine/context/context.go

```diff
@@ -146,7 +147,13 @@ func (ctx *context) AddRequest(request admissionv1.AdmissionRequest) error {
 }
 
 func (ctx *context) AddVariable(key string, value interface{}) error {
-	return addToContext(ctx, value, strings.Split(key, ".")...)
+	reader := csv.NewReader(strings.NewReader(key))
+	reader.Comma = '.'
```

Comment:
Can you add a testcase in `context_test.go` to show that this actually works now (so basically a test for `AddVariable`)? I am concerned that otherwise we might have a regression here eventually when the next person touches this code.